codec-http/src/main/java/io/netty/handler/codec/http/HttpMethod.java

```diff
@@ -143,18 +133,14 @@ private HttpMethod(String name, boolean bytes) {
             }
         }
 
-        this.name = name;
-        if (bytes) {
-            this.bytes = name.getBytes(CharsetUtil.US_ASCII);
-        } else {
-            this.bytes = null;
-        }
+        this.name = new AsciiString(name);
+        nameAsString = name;
     }
 
     /**
-     * Returns the name of this getMethod.
+     * Returns the name of this method.
      */
-    public String name() {
+    public AsciiString name() {
```

Comment:
@trustin - Thoughts...I'm move vested in the HttpHeaders discussion related to this, but it would be good to know especially before 4.1 is released.